lib/webauthn/authenticator_response.rb

```diff
@@ -25,15 +25,23 @@ def initialize(client_data_json:, relying_party: WebAuthn.configuration.relying_
     end
 
     def verify(expected_challenge, expected_origin = nil, user_presence: nil, user_verification: nil, rp_id: nil)
-      expected_origin ||= relying_party.origin || raise("Unspecified expected origin")
+      expected_origin ||= relying_party.allowed_origins || raise("Unspecified expected origin")
+
       rp_id ||= relying_party.id
 
       verify_item(:type)
       verify_item(:token_binding)
       verify_item(:challenge, expected_challenge)
       verify_item(:origin, expected_origin)
       verify_item(:authenticator_data)
-      verify_item(:rp_id, rp_id || rp_id_from_origin(expected_origin))
+
+      # NOTE: we are trying to guess from 'expected_origin' only in case it's a single origin
+      # (array that contains a single element)
+      # rp_id should either be explicitly set or guessed from only a single origin
```

Comment:
I feel I'd remove the comments we added for these methods. You did a good job on making them simple enough so they are easily understandable. While I know it's common practice to comment code we've followed the self documented code and only use comments and annotations for strictly necessary cases. I don't honestly think we're in that scenario here.